tiny_dnn/layers/recurrent_cell.h

```diff
@@ -146,15 +152,16 @@ class recurrent_cell_layer : public layer {
     params_.activation_ = std::shared_ptr<activation_layer>(activation);
   }
 
-  void init_backend(backend_t backend_type) {
-    CNN_UNREFERENCED_PARAMETER(backend_type);
-    core::OpKernelConstruction ctx =
-      core::OpKernelConstruction(layer::device(), &params_);
-    kernel_fwd_.reset(new RecurrentCellOp(ctx));
-    kernel_back_.reset(new RecurrentCellGradOp(ctx));
+  void init_backend(Device* device) {
```

Comment:
Now that it contains a ptr to the wrapper, I guess device is not needed as a parameter any more